standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2236,9 +2282,9 @@ struct GetSchemaResponse {
 
 struct GetPartitionRequest {
    1: optional string catName,
-   2: required string dbName,
-   3: required string tblName,
-   4: required list<string> partVals,
+   2: optional string dbName,
```

Comment:
why are these fields changed to optional? Shouldn't this request always be scoped to a table?

Reply:
2 reasons. 1) ease of use in HMSClient 2) Tests are sending null values in these required values and as a result, the request won't even land on HMS.
But both reasons are defeating the whole objective as these 3 values are always required. I'll handle the null values in the HMS client and keep these values as required.